src/Microsoft.AspNet.Mvc.Core/Filters/ProducesAttribute.cs

```diff
@@ -31,7 +32,13 @@ public override void OnResultExecuting([NotNull] ResultExecutingContext context)
 
             if (objectResult != null)
             {
-                SetContentTypes(objectResult.ContentTypes);
+                // Check if FormatFilter has already set the content type
```

Comment:
This code is now changed, we check if Format filter is present in pipeline and if it will set a content type. As it could be another Produces filter that has set the content type. We don't want to change it in that case.